### PR TITLE
Add User's validation of email

### LINKED ISSUE #45: Add validation for duplicate e-mail for user accounts
[Dependent on issue #41]

### PR DESCRIPTION
**Pull request by student-tomasz**
Closes #45

### CODE REVIEW

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+  it "is invalid with a blank password" do
+    expect(User.new(valid_user_params.merge!(password: ""))).not_to be_valid
+  end
+
+  it "is invalid with a password of less then 6 chars" do
+    expect(User.new(valid_user_params.merge!(password: "12345"))).not_to be_valid
+  end
+
+  it "is invalid with a password of more then 128 chars" do
```
Comment (agalkiewicz) [inline spec/models/user_spec.rb]: Maybe a good solution would be to divide test title. For example:

``` ruby
context "with a password of more...." do
  it "is invalid"
```

Or better:

``` ruby
context "is invalid" do
  it "when A" ...
  it "when B" ...
```

Comment (agalkiewicz) [inline spec/models/user_spec.rb]: It improves readability.

Comment (student-tomasz) [inline spec/models/user_spec.rb]: Yeah, I know. Never knew how to go about it. Your way has better readability when using the `-f d` rspec output, but then the pseudo-english in `_spec.rb` makes no sense.

Comment (AKorzeniowski) [inline spec/models/user_spec.rb]: How about dividing them also by what we're checking? I.e., 

``` ruby
context "invalid params" do
  context "password" do
    [password related checks]
```

**Comment by AKorzeniowski** [comment]
👍

**Comment by agalkiewicz** [comment]
:+1:

**Comment by gotar** [comment]
ok looks good. squash and merge

**Pull request merged**